mahara-autoscale-cache/scripts/helper_functions.sh

```diff
@@ -0,0 +1,765 @@
+#!/bin/bash
+
+# Common functions definitions
+
+function check_fileServerType_param
+{
+    local fileServerType=$1
+    if [ "$fileServerType" != "gluster" -a "$fileServerType" != "azurefiles" -a "$fileServerType" != "nfs" ]; then
+        echo "Invalid fileServerType ($fileServerType) given. Only 'gluster', 'azurefiles' or 'nfs' are allowed. Exiting"
+        exit 1
+    fi
+}
+
+function create_azure_files_mahara_share
+{
+    local storageAccountName=$1
+    local storageAccountKey=$2
+    local logFilePath=$3
+
+    az storage share create \
+        --name mahara \
+        --account-name $storageAccountName \
+        --account-key $storageAccountKey \
+        --fail-on-exist >> $logFilePath
+}
+
+function setup_and_mount_azure_files_mahara_share
+{
+    local storageAccountName=$1
+    local storageAccountKey=$2
+
+    cat <<EOF > /etc/mahara_azure_files.credential
+username=$storageAccountName
+password=$storageAccountKey
+EOF
+    chmod 600 /etc/mahara_azure_files.credential
+    
```

Comment:
still need these hardcoded endpoints fixed...

Reply:
Yes, I raised an issue about this, please see https://github.com/Azure/azure-quickstart-templates/issues/4960